### ISSUE TITLE
New Str encoding broke compatibility with 0.49 decoding

### ISSUE THREAD

**Issue by maxatome** [msgpack/msgpack-perl#30]
Since ada320e012380eb1006c020eaa7de8d48cd0843c the Bin encoding has been added.
So from this commit, utf8 strings are encoded using Str (old Raw) format and others with Bin one.
It appears that one can not choose to disable this new encoding and stay with the old Raw (Str) one.
It is problematic in a client/server environment, where Data::MessagePack versions are not necessarily the same at a given time, on both side...
I encounter this case (server sends binary non-utf8 strings using 0.50 and client uses 0.49), so unless a "str_as_bin" (or something like that) option is added, one have to stay with 0.49 version in such cases and not upgrade any more.

**Comment by syohex** [comment]
Yes, Data::MessagePack implements [new MessagePack specification](https://github.com/msgpack/msgpack/blob/master/spec.md) since version 0.50(Extension type is not supported yet). Please upgrade both server and client if you use version Data::MessagePack 0.50.

**Comment by maxatome** [comment]
It is too bad to not be able to keep the backward compatibility while taking advantage of bugs corrections of new versions... Sometimes one can not easily upgrade the client side (embedded environments for example) :(

**Comment by frsyuki** [comment]
Here is the guideline for libraries to support new format:
https://github.com/msgpack/msgpack/blob/master/spec.md#impl-upgrade

**Comment by syohex** [comment]
I released version 0.51 and 1.00. 0.51 reverts Str/Bin. If you want to use old specification please use it.

**Comment by dolmen** [comment]
For users of Data::MessagePack with Dist::Zilla, here is how to specify the prereq for the legacy msgpack behaviour:
```
[Prereqs]
Data::MessagePack = >= 0.36, < 1.00, != 0.49_01, != 0.50
```
